### PR TITLE
App install banner #427

### LINKED ISSUE #427: Add banner for mobile apps

### PR DESCRIPTION
**Pull request by naveed-ahmad**
This should resolved #427 

Opening PR for reviews and suggestions. Sorry still learning reactjs( and hating it so far ) so bear my bad ugly for now :)

### CODE REVIEW

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32941

**Comment by redhae** [comment]
Hi, 
i seem to be getting 2 banners on mobile safari

**Comment by naveed-ahmad** [comment]
@redhae thanks for reporting. Did you saw two banners simultaneously or closed the banner and saw again on page refresh ? 

Second case is known issue, working on that. But if first case if true, can you send screenshot or something to help me debugging ?

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32942

**Comment by sabeurthabti** [comment]
@naveed-ahmad amazing work bro. I don't have an android to test it on. Can you include some screenshots?

Happy for this to be merged in.

**Comment by naveed-ahmad** [comment]
@sabeurthabti preview for different mobiles and tables:

https://drive.google.com/file/d/0B-_ddcseNUwLeHZKYU84V3lzRUk/view

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32950

File: src/styles/components/SmartBanner.scss [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+  margin: 0 auto;
+  white-space: nowrap;
+}
+
+.smartbanner-close {
+  cursor: pointer;
+  display: inline-block;
+  vertical-align: middle;
+  margin: 0 5px;
+  font-family: 'ArialRoundedMTBold', Arial;
```
Comment (arshdkhn1) [inline src/styles/components/SmartBanner.scss]: IMO we should use font that we have used in quran.com. Font family should be either `source sans pro` or `Montserrat` or combination of both.

File: src/styles/components/SmartBanner.scss [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+/** Default **/
+.smartbanner {
+  position: absolute;
+  left: 0;
+  top: 0;
+  display: none;
+  width: 100%;
+  height: 80px;
+  line-height: 80px;
+  font-family: 'Helvetica Neue', sans-serif;
```
Comment (arshdkhn1) [inline src/styles/components/SmartBanner.scss]: IMO we should use font that we have used in quran.com. Font family should be either `source sans pro` or `Montserrat` or combination of both.

**Comment by arshdkhn1** [comment]
@naveed-ahmad https://egghead.io/lessons/react-react-fundamentals-development-environment-setup?course=react-fundamentals

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32952

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32953

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32954

**Comment by naveed-ahmad** [comment]
@arshdkhn1 removed the fonts, thanks for the course link.

**Comment by arshdkhn1** [comment]
@naveed-ahmad 
screenshot_2016-11-28-14-20-45-465_com android chrome
This is how banner looks in my mobile browser. 

Few suggestions - 

1- Quran logo is too blurry to be used. 

2- I see styling is different for different browsers/os. Did you change the colors of banner recently ?

3- IMO colors should be used that are complementary to our sites' color. They are looking a bit off the color scheme that is used in site.

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32971

**Comment by naveed-ahmad** [comment]
Android:
android-mobile

iOS:
ios-mobile

**Comment by ahmedre** [comment]
strange, the build is failing because of the favicon for some odd reason...

```
 [859] ../static/images/favicon.ico 215 bytes {0} [optional] [built] [failed] [1 error]
 [860] ../static/images/titles ^\.\/.*\.svg$ 3.35 kB {0} [built]
[1577] ../static/images ^\.\/.*$ 8.2 kB {0} [built]
[1578] multi main 40 bytes {0} [built]
    + 1654 hidden modules

WARNING in ../static/images/favicon.ico
Module parse failed: /quran/static/images/favicon.ico Unexpected character '' (1:0)
You may need an appropriate loader to handle this file type.
(Source code omitted for this binary file)
 @ ../static/images ^\.\/.*$
 @ ./components/SmartBanner/index.js
 @ ./containers/App/index.js
 @ ./routes.js
 @ ./client.js
 @ multi main
```

**Comment by ahmedre** [comment]
rebuild

**Comment by ahmedre** [comment]
out of curiosity, does this use whatever banner stuff Apple and Google have (i.e. will it detect if the app is installed and either not show the banner or show it with 'open' instead?)

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32972

**Comment by ahmedre** [comment]
img_1013

**Comment by naveed-ahmad** [comment]
@ahmedre it is using cookies to decide either to show banner or not. If user click on install then it won't show banner for 90 days, if user click close, then it will not show it for 15 days.

**Comment by ahmedre** [comment]
@naveed-ahmad imho, we should use whatever Apple and Google provide because they also know if the app is installed and don't bother the user. ex on iOS, i see double, and on Android, i have the app installed and see:

**Comment by naveed-ahmad** [comment]
@ahmedre what's your browser version ? Already heard some devices are showing the banner twice, looks like first is native banner and second one is custom one.

**Comment by ahmedre** [comment]
Safari on iOS 10.1.1

**Comment by ahmedre** [comment]
not to mention for Android, the native one is a lot nicer because it also pulls in info from Google Play and looks platform native - https://developer.android.com/distribute/users/banners.html

**Comment by naveed-ahmad** [comment]
For android, native banner is available only for Chrome 44 and  greater version:
https://developers.google.com/web/updates/2015/03/increasing-engagement-with-app-install-banners-in-chrome-for-android

And native iOS banners are available for safari 6 or greater.

**Comment by ahmedre** [comment]
can we do both then? native if we can, else fallback to this? sorry, i am not trying to be difficult!

**Comment by naveed-ahmad** [comment]
Yep working on it.

**Comment by mmahalwy** [comment]
@naveed-ahmad is this good to go?

**Comment by naveed-ahmad** [comment]
@mmahalwy No, working on showing native banner for advance browsers and fallback to custom banner for older browsers. Ill try to finish this tonight.

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32981

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32982

**Comment by mmahalwy** [comment]
good to go?

**Comment by naveed-ahmad** [comment]
@mmahalwy Gooood to go!

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32983

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32984

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32985

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32986

**Comment by mmahalwy** [comment]
good now? No more commits? :)

**Comment by naveed-ahmad** [comment]
@mmahalwy lol yeah good to go now.

**Pull request merged**